hail/src/main/scala/is/hail/expr/ir/ExtractIntervalFilters.scala

```diff
@@ -132,12 +136,12 @@ object ExtractIntervalFilters {
           case (None, None) =>
             None
         }
-      case ArrayFold(Literal(t, lit), False(), acc, value, body) =>
+      case ArrayFold(lit: Literal, False(), acc, value, body) =>
```

Comment:
I thought about checking here, but the interval filter will be MUCH more efficient than the existing linear search!